pkg/services/datasources/service/datasource_service.go

```diff
@@ -143,18 +143,21 @@ func (s *Service) GetDataSourcesByType(ctx context.Context, query *datasources.G
 func (s *Service) AddDataSource(ctx context.Context, cmd *datasources.AddDataSourceCommand) error {
 	return s.SQLStore.InTransaction(ctx, func(ctx context.Context) error {
 		var err error
-		// this is here for backwards compatibility
-		cmd.EncryptedSecureJsonData, err = s.SecretsService.EncryptJsonData(ctx, cmd.SecureJsonData, secrets.WithoutScope())
-		if err != nil {
-			return err
-		}
-
-		secret, err := json.Marshal(cmd.SecureJsonData)
-		if err != nil {
-			return err
+		if !s.features.IsEnabled(featuremgmt.FlagDisableSecretsCompatibility) {
+			cmd.EncryptedSecureJsonData, err = s.SecretsService.EncryptJsonData(ctx, cmd.SecureJsonData, secrets.WithoutScope())
+			if err != nil {
+				return err
+			}
+		} else {
+			cmd.EncryptedSecureJsonData = make(map[string][]byte)
```

Comment:
This is necessary because otherwise if EcryptedSecureJsonData is nil it's not set at all, this is replacing it by an empty map which ends up being set on the database.

Reply:
Actually, double checking on that.